kaitai/python/tests/utils.py

```diff
@@ -0,0 +1,205 @@
+# Utilities for tests cases for the python version of the Kaitai Struct SBP parser
+
+import kaitai.python.sbptable as kaitai_sbptable
+from kaitai.python.parse_utils import iter_messages as get_next_msg_kaitai, get_flattened_msg, serialise, BufferKaitaiStream
+from generator.sbpg.targets.common import snake_case, snake_case_keys, decode_json
+from kaitaistruct import KaitaiStream, KaitaiStruct
+import json
+import binascii
+import struct
+import sys
+import io
+import sbp.msg
+import sbp.table
+from sbp.msg import SBP_PREAMBLE
+import re
+from subprocess import Popen, PIPE
+
+SBP_HEADER_LEN = 6
+PERL_CMD = ['perl', 'kaitai/perl/bin/sbp2json.pl']
+
+
+# convert KaitaiStruct object into dict
+def dictify(obj, round_floats=False):
+    if isinstance(obj, KaitaiStruct):
+        return {k: dictify(v, round_floats) for k, v in obj.__dict__.items() if not k.startswith("_")}
+    elif isinstance(obj, dict):
+        return {k: dictify(v, round_floats) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        return [dictify(x, round_floats) for x in obj]
+    elif round_floats and type(obj) == float:
+        return float('{:.10f}'.format(obj)[:11])
+    else:
+        return obj
+
+
+# adapted from construct-based python version of sbp2json
+def parse_file_construct(fp):
```

Comment:
Worried that we're duplicating a lot of code here, can we just re-use the function directly?  Possible if we make adjustments to the original code we can just re-use directly here.

Reply:
Done